docs/source/faq.rst

```diff
@@ -0,0 +1,124 @@
+FAQ
+===
+
+In this section we grouped answers on frequently asked questions and some best practices of using `ignite`.
+
+
+Each engine has its own Events
+------------------------------
+
+It is important to understand that engines have their own events. For example, we defined a trainer and an evaluator:
+
+.. code-block:: python
+
+    @trainer.on(Events.EPOCH_COMPLETED)
+    def in_training_loop_on_epoch_completed(engine):
+        evaluator.run(val_loader) # this starts another loop on validation dataset to compute metrics
+
+    @evaluator.on(Events.COMPLETED)
+    def when_validation_loop_is_done(engine):
+        # do something with computed metrics etc
+        # -> early stopping or reduce LR on plateau
+        # or just log them
+
+
+Trainer engine has its own loop and runs multiple times over the training dataset. When a training epoch is over we
+launch evaluator engine and run a single time of over the validation dataset. **Evaluator has its own loop**. Therefore,
+it runs only one epoch and :attr:`~ignite.engine.Events.EPOCH_COMPLETED` is equivalent to :attr:`~ignite.engine.Events.COMPLETED`.
+As a consequence, the following code is correct too:
+
+.. code-block:: python
+
+    handler = EarlyStopping(patience=10, score_function=score_function, trainer=trainer)
+    evaluator.add_event_handler(Events.COMPLETED, handler)
+
+    best_model_saver = ModelCheckpoint('/tmp/models', 'best', score_function=score_function)
+    evaluator.add_event_handler(Events.COMPLETED, best_model_saver, {'mymodel': model})
+
+
+More details `Events and Handlers <concepts.html#events-and-handlers>`_.
+
+
+Creating Custom Events based on Forward/Backward Pass
+-----------------------------------------------------
+
+There are cases where the user might want to add events based on the loss calculation and backward pass. Ignite provides
+flexibility to the user to allow for this:
+
+.. code-block:: python
+
+    class BackpropagationEvents(Enum):
+        """
+        Events based on Backpropagation
+        """
+        BACKPROPAGATION_STARTED = 'backpropagation_started'
+        BACKPROPAGATION_ENDED = 'backpropagation_ended'
+        OPTIMIZER_STEP_END = 'optimizer_step_end'
+
+
+    def update(engine, batch):
+        model.train()
+        opitmizer.zero_grad()
+        x, y = process_batch(batch)
+        y_pred = model(x)
+        loss = loss_fn(y_pred, y)
+        engine.fire_event(BackpropagationEvents.BACKPROPOGATION_STARTED)
+        loss.backward()
+        engine.fire_event(BackpropagationEvents.BACKPROPOGATION_ENDED)
+        optimizer.step()
+        engine.fire_event(BackpropagationEvents.OPTIMIZER_STEP_END)
```

Comment:
```suggestion
        engine.fire_event(BackpropagationEvents.OPTIMIZER_STEP_ENDED)
```

Reply:
Although I did incorporate this, I'm rethinking this for this reason - https://github.com/pytorch/ignite/pull/471#discussion_r272749559